plugins/flashrom/meson.build

```diff
@@ -1,4 +1,5 @@
-if get_option('plugin_flashrom')
+if get_option('plugin_flashrom').enabled() or \
+   (get_option('plugin_flashrom').auto() and libflashrom.found())
```

Comment:
I'm not convinced this is the way we're supposed to use the meson feature thing. I think we're supposed to just use `if libflashrom.found()` -- no?

Reply:
I don't think so - because then you can end up with a case that flashrom is there but you set plugin_flashrom explicitly to disabled

Reply:
But the docs say `disabled -> do not look for the dependency and always return 'not-found'.` e.g. https://mesonbuild.com/Build-options.html

Reply:
oh!  that changes things.  the problem however is going to be when two things depend on it.  I can't see how you set required for two things.

Reply:
agree; i assume you can just look for the dep twice, the second being cached?

Reply:
I guess I'll need to experiment a little; it's ambiguous to me what happens.  Like what if you set redfish to enabled and curl to disabled but you happen to have the curl library around - does the daemon get curl support?  Or does it matter which call happened first?

Reply:
> does the daemon get curl support

I'd guess not, but I agree it's a bit underspecified.

Reply:
I think this is what is needed; https://mesonbuild.com/Reference-manual_returned_feature.html#featurerequire

Reply:
Be careful tho; that's `since 0.59.0` and we depend on 0.50.0 -- which might not be a problem, but might also be a can of worms.

Reply:
>  Like what if you set redfish to enabled and curl to disabled but you happen to have the curl library around - does the daemon get curl support? Or does it matter which call happened first?

If curl is disabled, and redfish is enabled, then the libcurl `lookup()` for curl is status "skipped" and does not get cached.

The libcurl lookup for redfish is status "look for it" (and also status "error if not found") so it progresses past the "skipped" early `return self._notfound_dependency()`, gets to the caching code, doesn't find a cached value, and does a full check.

If you have a third libcurl lookup that is auto (look for it, do not fail if not found) then it will do full lookup, cache the result, and persist that result for redfish too. But in that case, if it wasn't found then looking it up again was anyways going to fail.

tl;dr the cache is, or should be based on my reading of the implementation, safe. (If it isn't safe, that's a Meson bug. It is not *allowed* to have any effects other than speeding up successive lookups, so if it is doing something that is not allowed, and breaking the build as a result, that's a definite bug.)